### ISSUE TITLE
Things to do?

### ISSUE THREAD

**Issue by longemen3000** [juliacheme/PhysProps.jl#1]
for what i know, the utility of this package would be calling thermodynamic properties from the ChemSep Database. can i help in anything?

**Comment by braamvandyk** [comment]
That is right. We’ll limit this to pure components for now. Feel free to test and add features once the first running version is out.

**Comment by longemen3000** [comment]
i forked this and did an alternate implementation. the compound is stored in a `Dict{Symbol,Property}`, where property is a general wrapper for any property value (String, Float, Integer, Variable Property or Group Contribution), storing the long name, units and value.
the variable property is a struct that stores the parameters of a function and its range. is a callable struct, so you can do things like:
```julia
compound[:VaporPressure](150.0)
```

**Comment by braamvandyk** [comment]
I rather like the look of that. So far we have only finished the code to pull in the data from the database and populate the structs. The plan was to now write the functions that calculate the properties and this will be a lot easier to use than what I was heading to, i.e. VaporPressure(comp, T).

**Comment by longemen3000** [comment]
if you are using a Dict as a struct, you can use some pattern like:
```julia
function VaporPressure(comp,T)
  _prop = get(comp, :VaporPressure, nothing)
  isnothing(prop) && throw(error("there is no information on " * string(:VaporPressure)))
  val,in_range = value(_prop)(T)
  if in_range == false
    #do something to handle this (DomainError? a Default?)
  else
    return val
end
```
